pygenesys/utils/tsprocess.py

```diff
@@ -1,128 +1,107 @@
-        distribution_func = daily_hourly
-    return distribution_func
+    return pct, sorted_data
 
 
-def four_seasons_hourly(data_path, N_seasons=4, N_hours=24, kind='demand'):
+def get_peak_day(dataframe):
+    """
+    This function retrieves the peak day for a given timeseries.
     """
-    This function calculates a seasonal trend based on the
-    input data. Answers the question: what fraction of the annual
-    demand is consumed at this time of the year?
 
-    Parameters
-    ----------
-    data_path : string
-        The path to the time series data
-            * must be a ``.csv`` file
-            * nust have a column ``time`` that is a pandas datetime column
-        Tips:
-            * Sometimes a dataset will have an index column that can
-              be read as an ``Unnamed Column: 0``. If a user supplies
-              their own data, this should be removed where applicable.
-    kind : string
-        The string representing the kind of profile we're interested in.
-        Accepts: 'CF', 'cf', 'demand', 'Demand', 'DEMAND'
-            'CF' : A capacity factor profile is returned (sum != 1).
-            'Demand': Returns a demand profile (sum = 1)
-    N_seasons : integer
-        The number of seasons in the energy system model.
-    N_hours : integer
-        The hourly resolution of the energy system model.
+    time_series = dataframe.copy()
+    idx = time_series.where(
+        time_series == time_series.max()).dropna().index.date[0]
+    idx_delta = idx + dt.timedelta(days=1)
+    peak_day = time_series[idx:idx_delta][:24]
 
-    Returns
-    -------
-    distribution : numpy array
-        The time series data distributed over the specified time
-        slices.
+    return peak_day
+
+
+def get_weekends(dataframe):
+    """
+    This function retrieves the peak day for a given timeseries.
     """
-    time_series = pd.read_csv(data_path,
-                              usecols=[0, 1],
-                              index_col=['time'],
-                              parse_dates=True,
-                              )
 
-    # assumes northern latitudes
-    spring_mask = ((time_series.index.month >= 3) &
-                   (time_series.index.month <= 5))
-    summer_mask = ((time_series.index.month >= 6) &
-                   (time_series.index.month <= 8))
-    fall_mask = ((time_series.index.month >= 9) &
-                 (time_series.index.month <= 11))
-    winter_mask = ((time_series.index.month == 12) |
-                   (time_series.index.month == 1) |
-                   (time_series.index.month == 2))
+    time_series = dataframe.copy()
+    weekend_mask = (
+        time_series.index.weekday == 5) | (
+        time_series.index.weekday == 6)
+    weekends = time_series[weekend_mask]
+    weekend_hourly = weekends.groupby(weekends.index.hour).mean()[:24]
+
+    return weekend_hourly
+
+
+def get_season_masks(df):
+    """
+    Returns seasonal masks for given timeseries.
+    """
```

Comment:
This function (among others) could use the other numpy docstring elements, I know that they're going to be short because this function has one input and one output but when the docs are implemented it will be useful to have it already in place.